runtime/drivers/postgres/sql_store.go

```diff
@@ -0,0 +1,190 @@
+package postgres
+
+import (
+	"context"
+	sqldriver "database/sql/driver"
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/exp/slices"
+
+	// load pgx driver
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+// Query implements drivers.SQLStore
+func (c *connection) Query(ctx context.Context, props map[string]any) (drivers.RowIterator, error) {
+	sql, ok := props["sql"].(string)
+	if !ok {
+		return nil, fmt.Errorf("property \"sql\" is mandatory for connector \"motherduck\"")
+	}
+
+	res, err := c.db.QueryxContext(ctx, sql)
+	if err != nil {
+		return nil, err
+	}
+
+	schema, err := rowsToSchema(res)
+	if err != nil {
+		return nil, err
+	}
+
+	slices.SortFunc(schema.Fields, func(a, b *runtimev1.StructType_Field) bool {
+		return a.Name < b.Name
+	})
```

Comment:
Why is this needed? I think it will be unexpected for sources to have a different column ordering than the SQL statement

Reply:
I was previously relying on a different approach to scan data. Agreed its no longer required now with current approach.